public/app/features/alerting/unified/utils/cloud-alertmanager-notifier-types.ts

```diff
@@ -322,8 +322,12 @@ export const cloudNotifierTypes: Array<NotifierDTO<CloudNotifierType>> = [
         'The maximum number of alerts to include in a single webhook message. Alerts above this threshold are truncated. When leaving this at its default value of 0, all alerts are included.',
         {
           placeholder: '0',
+          inputType: 'number',
           validationRule: '(^\\d+$|^$)',
-          setValueAs: (value) => (typeof value === 'string' ? parseInt(value, 10) : 0),
+          setValueAs: (value) => {
+            const integer = Number(value);
+            return Number.isFinite(integer) ? integer : 0;
```

Comment:
is there a reason for not using the check?
```
  if (typeof value === 'string') {
    const integer = parseInt(value, 10);
    return isNaN(integer) ? 0 : integer;
  }
  return 0;
```

Reply:
Since I've also added a `type=number` in this PR there's no effective difference between `parseInt()` and `Number()` – though generally speaking the number constructor is a bit smarter and can correctly read hex and scientific notation (ie. `2e3`).

I usually prefer `Number.isFinite` over `isNaN` because I also don't want `+Infinity` and `-Infinity`, though again the number type input will prevent that from being passed in to the function anyway :D